File: Examples/LuaIntegration/luaScriptedBCAngledDuct/system/initBC.lua
print("Lua Init")

if faceStates and faceStates.refT and faceStates.state then
   print("Reading states")
   referenceT=faceStates.refT
   stateFace=faceStates.state
else
   print("Initializing states")
   referenceT=swak4foam.newScalarField(#refValue)
   for i=1,#referenceT do
      referenceT[i]=refValue[i]
   end

   stateFace=swak4foam.newScalarField(#refValue)
   for i=1,#referenceT do
      stateFace[i]=1
   end
   faceStates={
      refT=referenceT,
      state=stateFace
   }
end

File: Examples/Python3Integration/python3ScriptedBCAngledDuct/system/initBC.py
print("Python3 Init")

try:
    print("Reading states")
    referenceT=faceStates["refT"]
    stateFace=faceStates["state"]
except NameError:
    print("Initializing states")
    referenceT=refValue.copy()
    stateFace=numpy.zeros(refValue.size,dtype="i")
    faceStates={
        'refT'  : referenceT,
        'state' : stateFace
    }
